pkg/xds/sync/ingress_proxy_builder.go

```diff
@@ -57,7 +57,13 @@ func (p *IngressProxyBuilder) Build(key core_model.ResourceKey) (*xds.Proxy, err
 	if err != nil {
 		return nil, err
 	}
-	routing := p.resolveRouting(zoneIngress, zoneEgressesList, allMeshDataplanes, availableExternalServices)
+
+	allMeshGateways := &core_mesh.MeshGatewayResourceList{}
```

Comment:
Can you use these resources inside the `buildZoneIngressProxy` method, instead of fetching them again?

Reply:
I think the result is cached but yeah I switch the order around.